stl/inc/xutility

```diff
@@ -157,7 +173,12 @@ template <class _Ty>
 _Ty* __std_min_element(_Ty* _First, _Ty* _Last) noexcept {
     constexpr bool _Signed = _STD is_signed_v<_Ty>;
 
-    if constexpr (sizeof(_Ty) == 1) {
+    if constexpr (_STD is_same_v<_STD remove_const_t<_Ty>, float>) {
+        return const_cast<_Ty*>(static_cast<const _Ty*>(__std_min_element_f(_First, _Last, false)));

```

Comment:
I found that in order to fix #1596 (i.e. avoid ADL for `_Ty*`), we need to either `::`- (or equivalently `_CSTD`-) qualifiy `__std_min_element_f` or explicitly cast the pointers to `const void*`.

What's the preferred approach? (No change requested, as I'll do this in a new PR.)

Reply:
How would ADL work for pointers builtin types like `float` and `double` ?

Reply:
Oh, I've misread this. `__std_min_element_f` doesn't need to be qualified, but `__std_min_element_4`/`__std_min_element_8` may need because pointers may be passed to them as-is.

Reply:
I do not want to make this fix a part of this PR, it is a pre-existent bug with wider impact (find/count optimization is affected too)

Reply:
Updated, as it was updated in other occurrences